### DISCUSSION TITLE
[2023-06-05] Library and Engine Team Update

### DISCUSSION THREAD

**Discussion by jdunkerley** [enso-org/enso#6951] [Weekly Demo Day]
- James: Resolved the Regex issue and aligned with C#/Python. #6835 
- James: Added a load of ALIASes for functions. #6835 
- James: Better decode functions for web requests. #6835 
- James: Aligned with new dropdown FQN spec. #6835 
- James: Fixed issues with the table viz where `.` in the field name caused issues. #6933 
- James: Added a load of Locale and Encoding dropdowns. #6933 
- James: `set` checks the length when setting a column. #6933 
- James: Restructured the AWS credentials and started working towards S3 read.
- Radek: Added `Table.add_row_number`. #6890 
- Radek: Merged `Column_Index_Out_Of_Range` into `Missing_Input_Columns`. #6901
- Radek: Refactor `create_database_table` into `create_table` and `select_into_database_table`. #6925 
- Radek: Working on design and implementation for loading data into an existing database table.
- Greg: Added rounding functions to the Column type. #6817 
- Greg: Handle some edge cases in rounding due to in-exact floating point numbers. #6922 
- Greg: Reimplement `Column.truncate`, `ceil` and `floor` as vectorized ops. #6941 
- Jaroslav: Fix for runtime check of ascribed types works only for static methods. #6902 
- Jaroslav: Improve IGV Enso to recognize lib/scala/pkg as a project. #6904 
- Jaroslav: Adjust plugin so root of Enso repository as VSCode project. #6942
- Jaroslav: Fix to avoid swallowing exceptions in RuntimeServerTest. #6914
- Jaroslav: Fix for dates be dates and zones being zones. #6934
- Jaroslav: Documenting how to obtain static methods for a type. #6938 
- Jaroslav: Prototype work on hiding non-exported stuff from other libraries. #6905 
- Hubert: Fix so that empty project version is not treated as NaN. #6873 
- Hubert: Fix for dead lock scenario when doing file requests. #6920 
- Hubert: Engine throws a panic now (instead of an exception) on no currying for conversions. #6940 
- Hubert: Fix for potential NoSuchElementException. #6948 
- Pavel: Improve compiler diagnostic messages. #6931 
- Pavel: Working on detecting ambiguous imports and resolving the issues currently there. #6868 
- Dmitry: Consistent self types - fixing the static drop down issue. #6867 
- Dmitry: Suggestion response has a list of annotations. #6924
- Dmitry: Record the web socket requests. #6943 
- Dmitry: Fix for Atom constructor using two function calls. #6939 
- Dmitry: Create execution context with id. #6947

**Comment by radeusgd** [comment]
![image](https://github.com/enso-org/enso/assets/1436948/0c001c74-68d6-4c08-b9d2-3e80a9755374)
